lib/modules/eic_content/modules/eic_content_discussion/src/Hooks/EntityOperations.php

```diff
@@ -0,0 +1,217 @@
+        $contributorIds = [];
+
+        if (!$contributorsFieldList->isEmpty()) {
+          $this->contributors = $contributorsFieldList->referencedEntities();
+          $contributorIds = $this->getRelatedContributorIds();
+        }
+
+        if ($comment_contributorIds = $this->getDiscussionCommentContributorIds($entity)) {
+          foreach ($comment_contributorIds as $comment_contributorId) {
+            if (!in_array($comment_contributorId['uid'], $contributorIds)) {
+              $contributorIds[] = intval($comment_contributorId['uid']);
+            }
+          }
+        }
+
+        // Add contributor IDs to the renderable array.
+        $build['contributor_ids'] = $contributorIds;
+        break;
+
+    }
+  }
+
+  /**
+   * Helper function to get contributor IDs.
+   *
+   * @return array
+   *   Array of user IDs.
+   */
+  private function getRelatedContributorIds() {
+    $contributorUserIds = [];
+    foreach ($this->contributors as $contributorParagraph) {
+      $userRef = $contributorParagraph->get('field_user_ref');
+      $user = reset($userRef->referencedEntities());
+      // If Contributor is not a Platform Member, continue.
+      if ($user === FALSE) {
+        continue;
+      }
+      $contributorUserIds[] = intval($user->id());
+    }
+
+    return $contributorUserIds;
+  }
+
+  /**
+   * Helper function to add discussion owner as contributor.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  private function addOwnerAsContributor() {
+    $newContributorParagraph = Paragraph::create([
+      'type' => 'contributor',
+      'field_user_ref' => [
+        'target_id' => $this->discussion->getOwnerId(),
+      ],
+      'paragraph_view_mode' => 'platform_member',
+    ]);
+    $newContributorParagraph->save();
+
+    $contributors = [];
+    $contributors[] = $newContributorParagraph;
+    foreach ($this->contributors as $contributorParagraph) {
+      $contributors[] = [
+        'target_id' => $contributorParagraph->id(),
+        'target_revision_id' => $contributorParagraph->getRevisionId(),
+      ];
+    }
+
+    $this->contributors = $contributors;
+    $this->discussion->set('field_related_contributors', $contributors);
+  }
+
+  /**
+   * Helper function to get contributor IDs from discussion comments.
+   *
+   * @return array|bool
+   *   Array of user IDs or FALSE if no contributors have been found.
+   */
+  private function getDiscussionCommentContributorIds() {
+    $query = $this->database->select('comment_field_data', 'c')
+      ->fields('c', ['uid']);
+    $query->condition('c.entity_id', $this->discussion->id());
+    $query->condition('c.entity_type', 'node');
+    // Skip anonymous users.
+    $query->condition('c.uid', 0, '<>');
+    // We group by uid to avoid duplicated results.
+    $query->groupBy('c.uid');
+    $results = $query->execute()->fetchAll(\PDO::FETCH_ASSOC);
+
+    if (empty($results)) {
```

Comment:
What do you think about removing this "if" and directly return value ?

```suggestion
   return $results ?: FALSE;
```

Reply:
Nice one! Thanks :)